File: examples/key/new_managed_key_secret_based/new_managed_key_secret_based.py
from bloock.client.record import RecordClient
from bloock.client.key import KeyClient
from bloock.client.authenticity import AuthenticityClient
from bloock.entity.key.key_protection_level import KeyProtectionLevel
from bloock.entity.key.managed_key_params import ManagedKeyParams
from bloock.entity.key.key_type import KeyType
from bloock.entity.key.managed import Managed
from bloock.entity.key.access_control_secret import AccessControlSecret
from bloock.entity.authenticity.signer import Signer

# initialize the Key Client
key_client = KeyClient()
# initialize the Authenticity Client
authenticity_client = AuthenticityClient()
# initialize the Record Client
record_client = RecordClient()
# initialize the protection key level
protection = KeyProtectionLevel.SOFTWARE

# Generate a Rsa2048 key
key_type = KeyType.Rsa2048
key = key_client.new_managed_key(ManagedKeyParams(protection, key_type))

secret = "your secret"
email = "recovery@bloock.com"

# Create a Secret-Based access control for the created key
totp = key_client.setup_secret_access_control(Managed(key), secret, email)

record = record_client.from_string("Hello world").build()

# How to sign using the Secret-Based access control
access_control = AccessControlSecret(secret)
authenticity_client.sign(record, Signer(key, None, access_control))

File: examples/key/new_managed_key_secret_based/NewManagedKeySecretBased.java
import com.bloock.sdk.client.AuthenticityClient;
import com.bloock.sdk.client.KeyClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.authenticity.Signer;
import com.bloock.sdk.entity.key.AccessControl;
import com.bloock.sdk.entity.key.AccessControlSecret;
import com.bloock.sdk.entity.key.KeyProtectionLevel;
import com.bloock.sdk.entity.key.KeyType;
import com.bloock.sdk.entity.key.Managed;
import com.bloock.sdk.entity.key.ManagedKey;
import com.bloock.sdk.entity.key.ManagedKeyParams;
import com.bloock.sdk.entity.record.Record;

public class NewManagedKeySecretBased {
  public static void main(String[] args) throws Exception {
      // initialize the Key Client
      KeyClient keyClient = new KeyClient();
      // initialize the Authenticity Client
      AuthenticityClient authenticityClient = new AuthenticityClient();
      // initialize the Record Client
      RecordClient recordClient = new RecordClient();
      // initialize the protection key level
      KeyProtectionLevel keyProtectionLevel = KeyProtectionLevel.SOFTWARE;

      // Generate a Rsa2048 key
      KeyType keyType = KeyType.Rsa2048;
      ManagedKey key = keyClient.newManagedKey(new ManagedKeyParams(keyProtectionLevel, keyType));

      String secret = "your secret";
      String email = "recovery@bloock.com";

      // Create a Secret-Based access control for the created key
      keyClient.setupSecretAccessControl(new Managed(key), secret, email);

      Record record = recordClient.fromString("Hello world").build();

      // How to sign using the Secret-Based access control
      AccessControlSecret accessControl = new AccessControlSecret(secret);
      authenticityClient.sign(record, new Signer(key, null, new AccessControl(accessControl)));
  }
}
